Added the ban API now in a different way

pschichtel opened the pull request:
I added the new methods banPlayer(String name, String message), unbanPlayer(String name), banIp(String ip) and unbanIp(String ip) tp the Server class and added the new events BAN_PLAYER (called, when a player gets banned) and BAN_IP (called when an IP gets banned). The events are cancellable, so it is possible to abort a ban.

CraftBukkit pull-request: https://github.com/Bukkit/CraftBukkit/pull/266

Commit:
Fixed command aliases priorities.

Command aliases are now given lower weight than the "official" name of the command; this allows you to define aliases without worrying about them overriding someone else's command.

mikeprimm committed:
Add new Wolf methods for taming

verrier committed:
Added painting events (thanks verrier and tanelsuurhans)

Commit:
Add recipe classes and an addRecipe method in the server interface.

Rigby90 committed:
Added the ability to retrieve a world's Seed.

Commit:
Fixed AIOOBE when entering a space into console and hitting enter.

Commit:
Fixed getData() not returning the correct values.

grum committed:
Updated README for line-ending demands, also fixed line-endigs.

sunkid committed:
Added MaterialData support for SAPPLING, POWERED_RAIL, and DETECTOR_RAIL.

Commit:
Added a lot of events relating to weather, including those for entities. Thanks wizjany!

Raphfrk committed:
Added support for soft dependencies.

Soft dependencies allow plugins to request to be loaded after another plugin, but they will not throw an UnknownDependency exception if the other plugin is not present.

Commit:
Fixed formatting and removed tabs.

Raphfrk committed:
Improved the Scheduler.

Adds nag message when async tasks are not properly shut down and adds a limiter for sync tasks. Once they use 35ms in a single tick, any remaining tasks are not executed until later ticks. Adds a method to report the pending tasks and one to report active worker threads

Raphfrk committed:
Added setArmorContents as a method for PlayerInventory.

Rigby90 committed:
Added per world PVP setting support.

Commit:
Removed trailing spaces.

Commit:
Added SnowFormEvent. Thanks aPunch!

Commit:
Removed trailing spaces.

sk89q committed:
Added NoteBlock.play(instrument, note).

sk89q committed:
Added Player.playNote().

Commit:
Added PlayerInteractEntityEvent which fires when a player right clicks an entity. Thanks fullwall!

sk89q committed:
Added method to fake a block change request for a player. Alas, my Justin Bieber playing sign can work again!

Cogito committed:
AnimalTamer and Tameable interfaces, and corresponding features.
Documentation improvements.

grum committed:
Added javadoc generation, thanks lukegb

sk89q committed:
Added services manager framework. Services are interfaces that specifies capabilities to be implemented by providers. Example services include economy, <insert example 2>, etc.

sk89q committed:
Added Wolf.isTamed().

grum committed:
Missed site.xml for javadocs

sk89q committed:
Unadded (redundant) Wolf.isTamed().

tonybruess commented on src/main/java/org/bukkit/event/server/BanIpEvent.java:
```diff
@@ -0,0 +1,32 @@
+    public String getIp() {
+        return this.ip;
+    }
+
```
Maybe an option to see who sent the ban event?

pschichtel replied:
That's not possible here, but the plugins using the API can handle that.

tonybruess replied:
OK, sounds good.

pschichtel commented:
did anyone of the bukkit team review the pull by now?

Commit:
Added support for an update on load feature for plugins. Thanks Raphfrk!

Any files placed in the new (optional) update folder are automatically copied into the plugins directory the next time a reload happens. This allows safe updating of the plugin .jar files.

sk89q committed:
Changed LightningStrikeEvent to return a LightningStrike and not an Entity.

sk89q committed:
Added World.strikeLightningEffect() that does no damage.

sk89q committed:
Added dispenser event.

Cogito committed:
Added ability to saddle a pig to the API. Thanks xPaw!

Dinnerbone committed:
Line endings. Don't make me yell at you.

Dinnerbone committed:
Added @DontExport annotation for telling Bukkit not to share a class with another plugin

stevenh committed:
Fixed java plugin class loader so it works with plugins that contain classes also present in other plugins.
This also removes the changes from commit 1c4bde50bc12d130f6c8 which was added in order to fix this issue but wasn't ideal as it required plugins to be updated which isnt required with this fix

stevenh committed:
Guarded against plugins throwing exceptions on enable and disable to prevent potential corruption issues on server start and stop

grum committed:
src/site/apt/index.apt

sk89q committed:
Fixed errors during plugin enable/disable leaving Bukkit in an undefined state. Previous fix would at least prevent plugins from breaking the server, but it aborted the enable/disable process prematurely.

grum committed:
Whitespace + general cleanup

Commit:
Reverted commits that broke plugin class loading, pending investigation.

Commit:
Made disabled plugins show up in red in the plugins list. Thanks Yetanotherx!

Commit:
Added get/setLastDamageCause. Thanks sunkid!

stevenh committed:
Reworked command registration so that aliases are secondary to the primary command label requested
Updated docs to reflect what the code really does
This adds a few new methods to Command including:-
* A full constructor that takes descriptio, usageMassage and aliases for convenience
* getLabel() which returns the active label for a command, which is derived from the Command name or subsiquent call to setLabel(..)
* A number of registration functions for use in the CommandMap implementer
Also of note is Command.getAliases() no returns the "active" aliases

stevenh committed:
Added check and catch for errors during cancelTasks and unregister of services during plugin disable

stevenh committed:
Fixed no command found being sent to the command sender for command handlers which return failure (false)

Dinnerbone committed:
Added new World.Environment

Dinnerbone committed:
Added new Materials

pschichtel commented:
I'll implement this for 1.6

pschichtel commented:
fail, I closed the wrong pull request
